casbin/core_enforcer.py

```diff
@@ -178,11 +178,6 @@ def enable_enforce(self, enabled=True):
 
         self.enabled = enabled
 
-    def enable_log(self, enable):
-        """changes whether Casbin will log messages to the Logger."""
-
-        log.get_logger().enable_log(enable)
-
```

Comment:
Why is enable_log removed? We should try to keep the same usage as other languages.
https://github.com/casbin/casbin/blob/master/enforcer.go#L347

Reply:
We are removing **enable_log**, because if the root logger is not configured, no logs are going to be shown as you are using just INFO in your logs and the default level of root logger is WARNING, so it's the same as using that flag, that flag is not necessary anymore as we are just using the root logger.

> I think we should also implement the logger interface: https://github.com/casbin/casbin/blob/master/log/logger.go#L28
> 
> If you want to use the built-in `logging`, you can implement the interface yourself.

when a custom logger is created in a supporting module, we can have issues with it as the one that @jessecooper mentioned on https://github.com/casbin/pycasbin/issues/78

Reply:
We can implement `set_logger()` to customize logger, https://github.com/casbin/casbin/blob/master/enforcer.go#L173

Reply:
@techoner and @hsluoyz I can understand wanted to keep uniformity between the Casbin packages. We can add the arg back though it would be unneeded if you truly feel that strongly about it. The code would just be an unnecessary boolean check for the `enable_log` throughout the code. Also, having `set_logger` method would be an unnecessary method that would just set `self.logger = logging.getLogger()`. All of this will lead to slightly more bloated code and more inefficient code that I do not think is worth having in the name of uniformity. This is also a very slight deviation from uniformity. I think a package such as this should focus on being as efficient as possible for the language of implementation.

Reply:
@jessecooper @techoner @favipcj Currently, we have three options:

1. A more pythonic way like `logging.getLogger()` API.
2. Follow Casbin Golang's 3rd arg: `enable_log` in enforcer init.
3. Follow Casbin Golang's new logger interface: https://github.com/casbin/casbin/blob/master/log/logger.go#L28

I think definitely we can fulfill option 1 as we want to do it in a pythonic way internally as we are a Python package. And I think option 3 is currently unnecessary for this step.

But I think option 2 is OK to support. There will be not that much unnecessary and bloated code for it:

1. If the user sets `enable_log` to false, we can just do: (maybe wrong code here, just the meaning)

```py
self.tmp_logger = logging.getLogger()
logging.setLogger(None)
```

If the user sets `enable_log` to true:

```py
logging.setLogger(self.tmp_logger)
self.tmp_logger = None
```

2. For `set_logger()` API, we still let user do:

```py
logger = logging.getLogger()
enforcer.set_logger(logger)
```

Maybe this API is not that useful compared to Go, but it is at least making some sense that purely doing so:

```py
enforcer.set_logger()

def set_logger():
    self.logger = logging.getLogger()

```

Does this make sense for you?

Reply:
@hsluoyz would it be possible to add a footnote to this section of the docs stating:
> pycasbin leverages the default python logging mechanism and makes a call to `logging.getLogger()`. No special logging configuration is needed other than initializing the logger in the application.

I understand wanting things to be 1-to-1 on each implementation of Casbin but I do not think slight deviations will be all that detrimental to the project. Especially when it follows the best practices of the underlying language.

If you would like we can also make the modifications to the documentation if that is hosted within a repo on github and this is an acceptable path forward?

Reply:
@jessecooper OK. I think it's not a good idea to delay this PR any more. So I'm OK to follow your way for now and let's see if there are other customers feeling confused with our API or docs.

For docs, you can click "EDIT" in page: https://casbin.org/docs/en/log-error to make PR to the docs repo.

@favipcj Also plz resolve the conflicts first:

![image](https://user-images.githubusercontent.com/3787410/105563540-5e453b00-5d59-11eb-9259-6826b53abd58.png)